concepts/objects/about.md

```diff
@@ -0,0 +1,251 @@
+# About
+
+## Explanation
+
+Besides primitive data types like `number` and `string`, there is another important data type in JavaScript called `object`.
+Objects are collections of key-value pairs.
+In other languages they are referred to as maps or dictionaries and often times values need to have the same data type.
+In JavaScript, only the type of the key is restricted, it has to be a string.
+The values inside one object can have different types.
+They can be primitive types like numbers but also arrays, other objects or even functions.
+
+## Creating an Object
+
+You create an object (literal) with curly brackets.
+You can also directly include some entries (key-value pairs).
+For that, state the key first, followed by a colon and the value.
+
+```javascript
+const emptyObject = {};
+
+const obj = {
+  favoriteNumber: 42,
+  greeting: 'Hello World',
+  useGreeting: true,
+  address: {
+    street: 'Trincomalee Highway',
+    city: 'Batticaloa',
+  },
+  fruits: ['melon', 'papaya'],
+  addNumbers: function (a, b) {
+    return a + b;
+  },
+};
+```
+
+The trailing comma after the last entry is optional in JavaScript.
+
+You might wonder why the keys are not wrapped in quotation marks although they are supposed to be strings.
+This is a short-hand notation.
+If the key follows the naming rules for a JavaScript [identifier][mdn-identifier], you can omit the quotation marks.
+For keys with special characters in the name, you need to apply the usual string notation.
+
+```javascript
+const obj = {
+  '1keyStartsWithNumber': '...',
+  'key/with/slashes': '...',
+  'key-with-dashes': '...',
+  'key with spaces': '...',
+  '#&()[]{}èä樹keyWithSpecialChars': '...',
+};
+```
+
+Often times you want to combine existing variables into an object.
+This would often lead to key-value pairs like `name: name`.
+JavaScript has a short-hand notation that allows to just write `name` instead.
+There is also a short-hand notation for defining a function in an object.
+There you can omit the colon and the `function` keyword.
```

Comment:
Might not be the place to mention this, but the short-hand notation is **not** equivalent to the key-value notation. The difference is small, but there:

```javascript
const appleTree = {
  kind: 'apple',
  getParent: function() {
    return super.toString();
  }
};
// Uncaught SyntaxError: 'super' keyword unexpected here

const pearTree = {
  kind: 'pear',
  getParent() {
    return super.toString();
  }
};
// this works

// You can then declare things on its prototype
Object.setPrototypeOf(pearTree, { toString() { return "BigTree" } })
pearTree.getParent()
// => "BigTree"
```

It's a weird edgy case, but it matters as soon as you're going to type to _copy_ that method you've declared or if you're trying to write mixins.

Reply:
There's another _difference_ between these:

```javascript
const obj = {
  calcSum: function calc(a, b) {
    return a + b;
  },
};

obj.calcSum.name
// => "calc"

const obj2 = {
  calcSum: function (a, b) {
    return a + b;
  },
};

obj2.calcSum.name
// => "calcSum"

const obj3 = {
  calcSum(a, b) {
    return a + b;
  },
};

obj3.calcSum.name
// => "calcSum"
```

It's pretty insignificant, but 🤷🏽

Reply:
Re the first issue: 
Interesting, I didn't know and it was not mentioned [on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Method_definitions#description). Do you know why there is a difference or can you provide some link where I could read up on this? Tbh I would not hold off merging the PR until I got in. But I can put it into the design document as improvement idea. It might be a good fit once we have the special "sections" for about.md where you can mark something as advanced content.

Re the second issue:
True, but this is more about how function names are determined than about the shorthand notation imo. It could go into the functions concept but even there we have a lot more basic ground to cover first.

Its not worth holding off this PR for any of those. I would put both things into the design.md improvement ideas section of the respective exercise. They might be a good fit once we have the special "sections" for about.md where you can mark something as advanced content.